src/Middleware/HostFiltering/src/HostFilteringMiddleware.cs

```diff
@@ -142,37 +155,50 @@ private bool IsTopLevelWildcard(string host)
         }
 
         // This does not duplicate format validations that are expected to be performed by the host.
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         private bool CheckHost(HttpContext context, IList<StringSegment> allowedHosts)
         {
-            var host = new StringSegment(context.Request.Headers[HeaderNames.Host].ToString()).Trim();
+            var host = context.Request.Headers[HeaderNames.Host].ToString();
 
-            if (StringSegment.IsNullOrEmpty(host))
+            if (0u >= (uint)host.Length)
```

Comment:
Why not `0 == (uint)host.Length`?

Reply:
Weird...

Reply:
Let me recheck here

Reply:
Oh, ok. Its because the Jit likes `>=` tests on `uint`s to do bounds check elimination and the next test *was* checking if the first char was space; so it eliminated the first char bounds check.

However, that check is now gone, so can just do a regular zero test.

Reply:
Addressed